src/EditorFeatures/CSharpTest/Intents/DeleteParameterIntentTests.cs

```diff
@@ -0,0 +1,426 @@
+{
+    public static void DoFoo(this Foo foo, int value1, int value2)
+    {
+
+    }
+}";
+        var expectedText =
+@"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo(2);
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this Foo foo, int value2)
+    {
+
+    }
+}";
+
+        await VerifyExpectedTextAsync(WellKnownIntents.DeleteParameter, initialText, currentText, expectedText).ConfigureAwait(false);
+    }
+
+    [Fact]
+    public async Task TestDeleteParameterOnDefinitionAsync()
+    {
+        var initialText = @"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo(1, 2);
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this Foo foo, int {|priorSelection:value1|}, int value2)
+    {
+
+    }
+}";
+        var currentText =
+@"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo(2);
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this Foo foo, int value1, int value2)
+    {
+
+    }
+}";
+        var expectedText =
+@"public class Foo
+{
+    static void Bar()
+    {
+        var f = new Foo();
+        f.DoFoo(2);
+    }
+}
+
+public static class FooExtensions
+{
+    public static void DoFoo(this Foo foo, int value2)
+    {
+
+    }
+}";
+
+        await VerifyExpectedTextAsync(WellKnownIntents.DeleteParameter, initialText, currentText, expectedText).ConfigureAwait(false);
+    }
+
+    [Fact]
+    public async Task TestDeleteParamsParameterAsync()
+    {
+        var initialText =
+@"class C
+{
+    void M()
+    {
+        Method(new C(), {|priorSelection:1|}, 2, 3);
+    }
+
+    void Method(C c, params int[] values)
+    {
+    }
+}";
+        var currentText =
+@"class C
+{
+    void M()
+    {
+        Method(new C(), 2, 3);
+    }
+
+    void Method(C c, params int[] values)
+    {
+    }
+}";
+        var expectedText =
+@"class C
+{
+    void M()
+    {
+        Method(new C());
+    }
+
+    void Method(C c)
+    {
+    }
+}";
+
+        await VerifyExpectedTextAsync(WellKnownIntents.DeleteParameter, initialText, currentText, expectedText).ConfigureAwait(false);
+    }
+
+    [Fact]
+    public async Task TestDeleteParameterBeforeParamsAsync()
+    {
+        var initialText =
+@"class C
+{
+    void M()
+    {
+        Method(1.0f, 1, 2, 3);
+    }
+
+    void Method(float {|priorSelection:f|}, params int[] values)
+    {
+    }
+}";
+        var currentText =
+@"class C
+{
+    void M()
+    {
+        Method(1, 2, 3);
+    }
+
+    void Method(float f, params int[] values)
+    {
+    }
+}";
+        var expectedText =
+@"class C
+{
+    void M()
+    {
+        Method(1, 2, 3);
+    }
+
+    void Method(params int[] values)
+    {
+    }
+}";
+
+        await VerifyExpectedTextAsync(WellKnownIntents.DeleteParameter, initialText, currentText, expectedText).ConfigureAwait(false);
```

Comment:
so, i was also expecting that if we delete teh parameter from some constructor that we also remove any field/prop and/or assignemnt within the constructor.  Not part of this?

Reply:
Change signature doesn't support this today.  But if we did support that as part of change signature, this would work too since we're just calling into it.